Qute template detection has changed on Windows for templates in project dependencies

### Describe the bug

I have app that has Jakarta REST resources and Qute templates in separate Maven project and imports this project as dependency to Quarkus application. Now I can see that template is not detected anymore in 999-SNAPSOT. It seems like behavior has changed in last 24 hours. I have looked to merge commits and only thing that seems related by description is https://github.com/quarkusio/quarkus/pull/42492 (cc @gsmet ) but I didn't try it. I checked `io.quarkus.qute.deployment.QuteProcessor#collectTemplates` though and it seems like that PR touched related code.

### Expected behavior

Behavior on Linux and Windows should be same or differences should be documented.

### Actual behavior

Build fails and exception is raised:

```
2024-08-15T10:41:59.2320281Z [ERROR] Failed to execute goal io.quarkus:quarkus-maven-plugin:999-SNAPSHOT:build (build) on project qute-test: Failed to build quarkus application: io.quarkus.builder.BuildException: Build failure: Build failed due to errors
2024-08-15T10:41:59.2325699Z [ERROR] 	[error]: Build step io.quarkus.arc.deployment.ArcProcessor#generateResources threw an exception: jakarta.enterprise.inject.spi.DeploymentException: io.quarkus.qute.TemplateException: No template found for path [page] defined at parameter 'page' of io.quarkus.ts.qute.api.MyQuteResource constructor
2024-08-15T10:41:59.2327938Z [ERROR] 	- available templates: []
2024-08-15T10:41:59.2328556Z [ERROR] 	at io.quarkus.arc.processor.BeanDeployment.processErrors(BeanDeployment.java:1551)
2024-08-15T10:41:59.2329473Z [ERROR] 	at io.quarkus.arc.processor.BeanProcessor.processValidationErrors(BeanProcessor.java:187)
2024-08-15T10:41:59.2330356Z [ERROR] 	at io.quarkus.arc.deployment.ArcProcessor.generateResources(ArcProcessor.java:520)
2024-08-15T10:41:59.2331355Z [ERROR] 	at java.base/java.lang.invoke.MethodHandle.invokeWithArguments(MethodHandle.java:732)
2024-08-15T10:41:59.2332160Z [ERROR] 	at io.quarkus.deployment.ExtensionLoader$3.execute(ExtensionLoader.java:854)
2024-08-15T10:41:59.2332913Z [ERROR] 	at io.quarkus.builder.BuildContext.run(BuildContext.java:256)
2024-08-15T10:41:59.2333699Z [ERROR] 	at org.jboss.threads.ContextHandler$1.runWith(ContextHandler.java:18)
2024-08-15T10:41:59.2334923Z [ERROR] 	at org.jboss.threads.EnhancedQueueExecutor$Task.doRunWith(EnhancedQueueExecutor.java:2516)
2024-08-15T10:41:59.2336112Z [ERROR] 	at org.jboss.threads.EnhancedQueueExecutor$Task.run(EnhancedQueueExecutor.java:2495)
2024-08-15T10:41:59.2336962Z [ERROR] 	at org.jboss.threads.EnhancedQueueExecutor$ThreadBody.run(EnhancedQueueExecutor.java:1521)
2024-08-15T10:41:59.2337650Z [ERROR] 	at java.base/java.lang.Thread.run(Thread.java:840)
2024-08-15T10:41:59.2338195Z [ERROR] 	at org.jboss.threads.JBossThread.run(JBossThread.java:483)
2024-08-15T10:41:59.2339710Z [ERROR] Caused by: io.quarkus.qute.TemplateException: No template found for path [page] defined at parameter 'page' of io.quarkus.ts.qute.api.MyQuteResource constructor
2024-08-15T10:41:59.2341103Z [ERROR] 	- available templates: []
2024-08-15T10:41:59.2341844Z [ERROR] 	at io.quarkus.qute.deployment.QuteProcessor.validateTemplateInjectionPoints(QuteProcessor.java:2241)
2024-08-15T10:41:59.2342526Z [ERROR] 	... 9 more
2024-08-15T10:41:59.2342807Z [ERROR] -> [Help 1]
```


### How to Reproduce?

Steps to reproduce the behavior:

1. `git clone git@github.com:quarkus-qe/quarkus-test-suite.git`
2. `cd quarkus-test-suite/qute/multimodule`
3. `mvn clean verify`

### Output of `uname -a` or `ver`

Microsoft Windows Server 2022 10.0.20348

### Output of `java -version`

Temurin 17.0.12+7

### Quarkus version or git rev

999-SNAPSHOT

### Build tool (ie. output of `mvnw --version` or `gradlew --version`)

Apache Maven 3.8.7 (b89d5959fcde851dcb1c8946a785a163f14e1e29)

### Additional information

You can see logs here: https://github.com/quarkus-qe/quarkus-test-suite/actions/runs/10401726443/job/28805582860
And previous runs here: https://github.com/quarkus-qe/quarkus-test-suite/actions/workflows/daily.yaml

--- a/independent-projects/bootstrap/app-model/src/main/java/io/quarkus/paths/OpenContainerPathTree.java
+++ b/independent-projects/bootstrap/app-model/src/main/java/io/quarkus/paths/OpenContainerPathTree.java
@@ -93,7 +93,7 @@ public abstract class OpenContainerPathTree extends PathTreeWithManifest impleme
 
     @Override
     public Collection<Path> getRoots() {
-        return List.of(getContainerPath());
+        return List.of(getRootPath());
     }
 
     @Override

--- a/independent-projects/bootstrap/app-model/src/main/java/io/quarkus/paths/PathTree.java
+++ b/independent-projects/bootstrap/app-model/src/main/java/io/quarkus/paths/PathTree.java
@@ -83,8 +83,7 @@ public interface PathTree {
     /**
      * The roots of the path tree.
      * <p>
-     * Note that for archives, it will return the path to the archive itself,
-     * not a path that you can browse.
+     * Note that you shouldn't use these roots for browsing except if the PathTree is open.
      *
      * @return roots of the path tree
      */

